PROTOCOL.md

```diff
@@ -137,16 +137,16 @@ For the sake of clarity, the following examples demonstrate the communication pr
 
 <h3 id="successful-connection-initialisation">Successful connection initialisation</h3>
 
-1. _Client_ sends a WebSocket handshake request with the sub-protocol: `graphql-transport-ws`
-1. _Server_ accepts the handshake and establishes a WebSocket communication channel (which we call "socket")
-1. _Client_ immediately dispatches a `ConnectionInit` message optionally providing a payload as agreed with the server
+1. _Client_ sends a WebSocket handshake request through HTTP with the sub-protocol: `graphql-transport-ws`
+1. _Server_ evaluates the request, checks the supported subprotocols and concludes the upgrade with an `101: Switching Protocols` response so accepts the handshake and establishes a WebSocket communication channel (which we call "socket")
```

Comment:
Server should accept the connection if the WS subprotocol is `graphql-transport-ws`. The "checks the supported subprotocols" part is superfluous because the previous step already explains that you're requesting an upgrade with the right subprotocol. Furthermore, what you do with invalid subprotocols is completely out of this Protocol's scope, it is exclusive to `graphql-transport-ws` subprotocol.

Also, `101: Switching Protocols` is part of the WebSocket standard on how to upgrade the transports. There is no need to repeat ourselves.

Reply:
great point
for these reasons I am closing this PR